src/platform/with_wgl/utils.rs

```diff
@@ -25,60 +25,61 @@ use super::wgl_ext;
 pub unsafe fn create_offscreen(shared_with: winapi::HGLRC,
                                settings: &WGLAttributes)
                                -> Result<(winapi::HGLRC, winapi::HDC), String> {
-
-    let window: winapi::HWND = try!(create_hidden_window());
-    let hdc = user32::GetDC(window);
-    if hdc.is_null() {
+    let mut ctx = WGLScopedContext::default();
+    ctx.window = try!(create_hidden_window());
+    ctx.device_ctx = user32::GetDC(ctx.window);
+    if ctx.device_ctx.is_null() {
         return Err("GetDC function failed".to_owned());
     }
 
-    let extra = try!(load_extra_functions(window));
+    let extra = try!(load_extra_functions(ctx.window));
 
     let extensions = if extra.GetExtensionsStringARB.is_loaded() {
-        let data = extra.GetExtensionsStringARB(hdc as *const _);
+        let data = extra.GetExtensionsStringARB(ctx.device_ctx as *const _);
         let data = CStr::from_ptr(data).to_bytes().to_vec();
         String::from_utf8(data).unwrap()
-
     } else if extra.GetExtensionsStringEXT.is_loaded() {
         let data = extra.GetExtensionsStringEXT();
         let data = CStr::from_ptr(data).to_bytes().to_vec();
         String::from_utf8(data).unwrap()
-
     } else {
         String::new()
     };
 
-
     let (id, _) = if extensions.split(' ').find(|&i| i == "WGL_ARB_pixel_format").is_some() {
-        try!(choose_arb_pixel_format(&extra, &extensions, hdc, &settings.pixel_format)
-            .map_err(|_| "Pixel format not available".to_owned()))
+        try!(choose_arb_pixel_format(&extra, &extensions, ctx.device_ctx, &settings.pixel_format)
+            .map_err(|_| "ARB pixel format not available".to_owned()))
     } else {
-        try!(choose_native_pixel_format(hdc, &settings.pixel_format)
-            .map_err(|_| "Pixel format not available".to_owned()))
+        try!(choose_native_pixel_format(ctx.device_ctx, &settings.pixel_format)
+            .map_err(|_| "Native pixel format not available".to_owned()))
     };
 
-    try!(set_pixel_format(hdc, id));
-
-    create_full_context(settings, &extra, &extensions, hdc, shared_with)
+    try!(set_pixel_format(ctx.device_ctx, id));
 
+    let result = create_full_context(settings, &extra, &extensions, ctx.device_ctx, shared_with);
+    if result.is_ok() {
+        ctx.dispose = false; // Everything is ok, don't dispose the scoped ctx
```

Comment:
probably something like removing the `dispose` flag, and use `mem::forget` is better here? It's probably fine anyway.